portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,12 +39,22 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
+    String strNumComments = request.getParameter("numComments");
+    int numComments;
+    try{ 
+      numComments = Integer.parseInt(strNumComments);
+    } catch (Exception e) { 
+      numComments = 10;
```

Comment:
it's generally going to be better to return an error in the case of bad arguments for an API like this. Otherwise, silent failures are often overlooked.